### ISSUE TITLE
JS exception in pre-need burial application due to missing items in uiSchema

### ISSUE THREAD

**Issue by mbmasuda** [department-of-veterans-affairs/va.gov-team#25109]
# I am not a VA employee—please do whatever needs to be done to put this bug report in the right place

Tagging @1Copenut and @timwright12 because it seems they made the most recent [commit to the burials portion](https://github.com/department-of-veterans-affairs/vets-website/commits/master/src/applications/burials)

## What happened?

On the last step (6 of 6) before submission on https://www.va.gov/burials-and-memorials/pre-need/form-10007-apply-for-eligibility/review-and-submit when I click on the [+] to expand the "Military History" section (this is the only section with this issue) to double check the info, it POSTs a request with the following headers (I deleted my cookies from the copy/paste, obviously)

```
POST /js-report/api/0/store/?sentry_key=faker&sentry_version=7 HTTP/1.1
Host: www.va.gov
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:88.0) Gecko/20100101 Firefox/88.0
Accept: */*
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate, br
Referer: https://www.va.gov/
Content-Type: text/plain;charset=UTF-8
Origin: https://www.va.gov
Content-Length: 26261
DNT: 1
Connection: keep-alive
Cookie: <redacted>
```

and the console shows an issue with processing the response JSON that occurs in utils.js:314:12:

```
Uncaught Error: uiSchema order list does not contain properties 'dateRange', 'dischargeType', 'highestRank', 'nationalGuardState'
```

The actual response headers are:

```
HTTP/1.1 200 OK
Date: Wed, 05 May 2021 14:59:25 GMT
Content-Type: application/json
Connection: keep-alive
Access-Control-Allow-Headers: X-Sentry-Auth, X-Requested-With, Origin, Accept, Content-Type, Authentication, Authorization
X-Content-Type-Options: nosniff
Content-Language: en
Access-Control-Expose-Headers: X-Sentry-Error, Retry-After
Expires: Wed, 05 May 2021 14:59:25 GMT
Vary: Accept-Language, Cookie
X-XSS-Protection: 1; mode=block
Allow: GET, POST, HEAD, OPTIONS
Cache-Control: no-cache, no-store, must-revalidate, max-age=0
Access-Control-Allow-Origin: https://www.va.gov
Access-Control-Allow-Methods: GET, POST, HEAD, OPTIONS
X-Frame-Options: deny
Content-Security-Policy-Report-Only: script-src 'self' 'unsafe-inline' 'nonce-88X7LTHDMlKrFDb7FLVMStuh4cFD9pdc' http://www.google-analytics.com https://*.uservoice.com https://dap.digitalgov.gov https://designsystem.digital.gov https://maps.googleapis.com https://standards.usa.gov https://www.google-analytics.com https://www.googletagmanager.com https://tagmanager.google.com 'unsafe-eval' https://optimize.google.com https://gateway.foresee.com https://resources.digital-cloud-gov.voice.medallia.com https://prod-va-gov-assets.s3-us-gov-west-1.amazonaws.com https://www.youtube.com https://*.ytimg.com https://cdn.botframework.com 'strict-dynamic' nonce-88X7LTHDMlKrFDb7FLVMStuh4cFD9pdc; img-src 'self' data: blob: https://*.gstatic.com https://api.mapbox.com https://www.google-analytics.com https://www.googletagmanager.com https://stats.g.doubleclick.net https://*.va.gov https://optimize.google.com https://gateway.foresee.com https://static.foresee.com https://cdn-prod.kampyle.com https://prod-va-gov-assets.s3-us-gov-west-1.amazonaws.com https://ok6static.oktacdn.com https://dvp-oauth-application-directory-logos.s3-us-gov-west-1.amazonaws.com ; frame-ancestors 'none' ; manifest-src 'none' ; default-src 'self' ; frame-src https://dap.digitalgov.gov ; child-src https://*.uservoice.com https://www.googletagmanager.com https://www.youtube.com https://optimize.google.com https://resources.digital-cloud-gov.voice.medallia.com ; style-src 'self' 'unsafe-inline' https://fonts.googleapis.com https://tagmanager.google.com https://optimize.google.com https://fonts.googleapis.com https://gateway.foresee.com https://medallia.s3.amazonaws.com https://prod-va-gov-assets.s3-us-gov-west-1.amazonaws.com ; connect-src 'self' http://localhost:4000 https://*.va.gov https://api.mapbox.com https://www.google-analytics.com http://*.vetsgov-internal https://prod-va-gov-assets.s3-us-gov-west-1.amazonaws.com https://prod-va-gov-maintenance-windows.s3-us-gov-west-1.amazonaws.com https://analytics.foresee.com https://brain.foresee.com https://survey.foreseeresults.com https://device.4seeresults.com https://health.foresee.com https://gateway.foresee.com https://feedback.digital-cloud-gov.voice.medallia.com https://raw.githubusercontent.com wss://northamerica.directline.botframework.com https://northamerica.directline.botframework.com https://search.usa.gov ; object-src 'self' blob: ; media-src 'none' ; font-src 'self' data: https://fonts.gstatic.com https://gateway.foresee.com https://medallia.s3.amazonaws.com https://prod-va-gov-assets.s3-us-gov-west-1.amazonaws.com ; block-all-mixed-content  ; form-action 'self' https://vicbdc.vba.va.gov https://vaww.vicbdc.ppd.vba.va.gov https://feedback.digital-cloud-gov.voice.medallia.com ; base-uri http://*.vetsgov-internal https://*.va.gov https://optimize.google.com ;
Referrer-Policy: no-referrer-when-downgrade
Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; report=/csp-report
Content-Encoding: gzip
Transfer-Encoding: chunked
```

## Specs:
- Device: Macbook Pro
- Browser: Firefox
- Test User _(if applicable)_:  I am a veteran, not a VA employee

## Steps to Reproduce

See above description.  Basically, fill out the pre-need burial application and on the final confirmation page, try to expand the "Military History" section and it will trigger the exception when the AJAX response is received.

## Desired behavior
_Describe in detail what was supposed to happen; reference existing Github issue that contains a solution if possible_

It should not crash, the "Military History" should be displayed with correct info, and the form should be able to be submitted.

## Acceptance Criteria
- [ ] _Describe the desired working behavior_
- [ ] The correct "Military History" info should be displayed
- [ ] The form can be submitted successfully

## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to Epic** (what body of work is this a part of? possibly `Ongoing Maintenance`)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `triage`, `tools-improvements`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with `Bug`**

**Comment by timwright12** [comment]
@mbmasuda Thanks! I'll pass this along to the relevant engineers

**Comment by mbmasuda** [comment]
Hi, I was not sure if I did a good job of making it clear that this bug causes a rendering crash and it is not possible to verify the Military History info, and due to the rendering crash, one cannot submit the form after triggering the issue. I suppose I could just submit the form blindly without double-checking my info, but I didn't want to do that so I am just going to send the application in the mail.

Here is a screen movie of what happens to me. I am on Mac Big Sur 11.4; Firefox 89.0; Mozilla VPN in use.

https://user-images.githubusercontent.com/1646132/121992242-1baaec00-cd56-11eb-9c99-4288a0539f00.mp4

**Comment by Mottie** [comment]
Fixed in production

**Comment by mbmasuda** [comment]
Hi, sorry but I need to reopen this issue because the fix did not seem to work for me. I am using Big Sur 11.4 with Firefox 89.0.2 in the following screen movie.

https://user-images.githubusercontent.com/1646132/124148654-b1c45f00-da44-11eb-8823-befc4840617a.mp4

**Comment by mbmasuda** [comment]
Tagging @Mottie and @ashagross since I forgot to do that in my previous comment.

**Comment by chrisj-usds** [comment]
Hi @Mottie -- Is there any update or status on this issue? It's come back around with various folks at VA as part of a larger problem.

**Comment by Mottie** [comment]
I haven't had a chance to look into this within the past 7 days. I will try to dig into it again today.

**Comment by Mottie** [comment]
@mbmasuda We're not able to duplicate the same error. If you could provide us with the application ID number (in bold) from within the green box, we might be able to get a better idea of where to look for the problem.

![](https://user-images.githubusercontent.com/136959/124982382-3f332080-dffc-11eb-8328-c166d1a43a31.png)

**Comment by mbmasuda** [comment]
Sure, I am not right in front of my computer at the moment but will get the info when I get home.

**Comment by mbmasuda** [comment]
Hi, I got home and just tried to log in to va.gov using my DS Login to try to get the ID you are requesting, but now I cannot log in to va.gov due to the following issue:

<img width="1792" alt="Screen Shot 2021-07-08 at 17 59 19" src="https://user-images.githubusercontent.com/1646132/125008190-d9a05d80-e016-11eb-8d99-f8124ca29e95.png">

<img width="1792" alt="Screen Shot 2021-07-08 at 17 59 53" src="https://user-images.githubusercontent.com/1646132/125008464-6d722980-e017-11eb-840f-11b5d33b0a79.png">

Based on the source in the second screenshot, it appears the HTML template in use is super old, since using a CDATA section inside a `<script>` tag is only necessary when the document is being parsed as XHTML and not HTML5. Based on the response's content type of `text/html` I do not think the CDATA section is necessary for this particular use, but I do not know your template architecture, so blindly updating that could break something else if the template is also used by other stuff that is still being served as XHTML.

Here is a portion of the response showing a content type of `text/html`:

<img width="1680" alt="Screen Shot 2021-07-08 at 18 10 54" src="https://user-images.githubusercontent.com/1646132/125008812-27699580-e018-11eb-8add-9a0e07596539.png">


That issue aside, I am pretty sure the reason "big VA" is asking about this is because I was in contact with the acting DCIO for account management (I don't want to name names in a public forum), and the reason I contacted that person was related to the fact that a letter addressed to me went to a random address in a state I have never lived in. I did speak on the phone with the supervisor of the unit that mailed the letter and I have a theory of what may have happened on the backend, but again, I don't want to post that info in a public forum. If someone can reach out to me I can explain the things I experienced and found out from the person I spoke to on the phone to help troubleshoot this issue, but I don't want to put all the details on GitHub for the world to see.

I have a plumber coming over tomorrow (Friday, July 9, 2021) morning to do some stuff, but hopefully they will finish around noon or so Pacific time. I can speak to someone on the phone after then, or next week if that is better for your schedule.

Thanks!

**Comment by mbmasuda** [comment]
Hi,

I was able to log in to va.gov with my DS Login right now so I guess whatever went wrong the other day with the login is now working. 🎉 I did notice that it seems all login options eventually pass through api.id.me, and I am not sure if maybe the glitch from the other day was with id.me.

I think the issue with my pre-need burial application as far as I can tell is the node in the DOM that should display the ID you are requesting is totally empty for me so it does not display anything on the UI. This might be because my application has already been processed. I don't want to actually submit this form because they already received two online applications and a paper application from me due to the mixup about the response letter being sent to the wrong address so I did not know they had already responded way back in the beginning of April. I could wait until my current in-progress application expires and then start a new one (with no intention of actually submitting it) to see if the issue is that my current application has already been processed, or if there is some other issue.

#### CSS selector path of blank DOM node
```
#react-root > div > div:nth-child(2) > div:nth-child(1) > div > div.progress-box.progress-box-schemaform > div > div:nth-child(5)
```

#### Same node as above but as XPath path
```
//*[@id="react-root"]/div/div[2]/div[1]/div/div[4]/div/div[5]
```

Here is what the DOM inspector looks like in Safari when the page is loaded without any manual changes to the DOM:

![Screen Shot 2021-07-12 at 05 43 19](https://user-images.githubusercontent.com/1646132/125289617-4b58ff80-e2d4-11eb-973d-46226eebb1b3.png)


Here is what the DOM inspector looks like in Safari when the page is loaded with my fake debugging message changes to the DOM:

![Screen Shot 2021-07-12 at 05 41 59](https://user-images.githubusercontent.com/1646132/125289758-73e0f980-e2d4-11eb-811a-a995b9c91cf7.png)

Out of curiosity, when you tested this and could not reproduce what I was seeing, were you testing on a Mac? I was originally using Firefox, but I took the above screenshots with Safari, so it could be some Mac-specific issue, and if it is, please test if the problem only occurs on Big Sur or if Catalina and lower also have this issue. Thanks!

**Comment by Mottie** [comment]
I did try to duplicate this issues on a Mac using Catalina & Chrome without success. To get the green box ID to show up, make a change within the form. Sharing this will provide us with the save-in-progress data which may or may not allow us to reproduce the issue you're describing.

**Comment by mbmasuda** [comment]
Ok, I will try that but unfortunately my power went out so my internet is down and my cell reception is really terrible in my place so I can't use my phone as a tether. When the power comes back on, I'll make a change in the form on my computer and reply with the results. Thanks!

P.S. I originally started this application in late March 2021 so if there were any schema changes in the meantime, perhaps my application data no longer matches the currently expected format. 🤷🏻‍♀️

**Comment by mbmasuda** [comment]
My power came back on and the ID that appeared after I did a small edit to my in-progress application is `2881095`, however, the supervisor I spoke to on the phone emailed me a PDF of the letter from April, and the ID on the PDF is `212419`. I did not receive the paper letter in the mail yet, but when it comes, I will make sure the ID on that is the same as what is on the PDF. Thanks!

**Comment by mbmasuda** [comment]
Hi, since you mentioned you could not duplicate the issue on Catalina with Chrome, I downloaded Chrome and it still did not work for me even after I did the small edit to cause the ID number to show up. Therefore, the issue could be something to do with a security update or other change between Catalina and Big Sur.

I have been unsuccessful on the three major browsers—FIrefox 89.0.2; Safari 14.1.1 (16611.2.7.1.4); and Chrome (whatever version was available today, but I already deleted it because I don't need Google spying on me and I forgot to get the version number)—with Big Sur 11.4, so if you can please try it on your end with a Mac running Big Sur 11.4 then if it still does not reproduce for you, then that can help to narrow down the potential causes.

I could not see an obvious way for me to start a new pre-need burial application so I am not sure how I can check from my side if the problem is related to a schema change, because when I submitted the application on March 20 and March 25, the form looked different.

**Comment by Mottie** [comment]
Thank you. I am able to reproduce the problem with your data. I will spend some time troubleshooting it today.

**Comment by mbmasuda** [comment]
👍🏼

To be clear, my personal situation is already resolved because the supervisor sent me a PDF of the letter that initially went to an address in a state I've never lived in, but if the root cause of this issue is somehow related to that issue, then finding out the problem could help fix that issue for other people.

**Comment by Mottie** [comment]
Hi @mbmasuda! Thanks for your help and feedback. This problem should now be fixed. Please let us know if you still encounter any problems.

**Comment by mbmasuda** [comment]
🎉🎉🎉 Awesome! Thanks for the fast fix! I did just log in to my va.gov account and was able to expand the "Military History" block and it did not have a rendering crash any more and the info I had entered previously was displayed. I did not actually submit the form, though, since I already received the approval letter and don't want to cause confusion on the other end by submitting my info yet again.

Since it seems to be working now, I am closing this issue.
